### PR TITLE
fix(std/node): Add ascii support for Buffer

### PR DESCRIPTION
**Pull request by Soremwar**
Towards merging https://github.com/nodejs/readable-stream/pull/452

### CODE REVIEW

File: std/node/buffer.ts [hunk 1]
```diff
@@ -4,13 +4,22 @@ import * as base64 from "../encoding/base64.ts";
...
 const notImplementedEncodings = [
-  "ascii",
   "binary",
   "latin1",
   "ucs2",
   "utf16le",
 ];
 
+function decodeAscii(buffer: Uint8Array) {
+  return Array.from(buffer).map((byte) => String.fromCharCode(byte & 0x7F)).join("");
```
Comment (benjamingr) [inline std/node/buffer.ts:14]: Any reason this these aren't done with `TextEncoder` / `TextDecoder` ?

Comment (benjamingr) [reply std/node/buffer.ts:14]: Or encoding/ascii directly if that's an issue

Comment (Soremwar) [reply std/node/buffer.ts:14]: AFAIK TextEncoder is UTF8 only

File: std/node/buffer.ts [hunk 2]
```diff
@@ -4,13 +4,22 @@ import * as base64 from "../encoding/base64.ts";
...
+}
+
+function encodeAscii(text: string) {
+  return new Uint8Array(
+    Object.keys(text).map((i) => text.charCodeAt(Number(i))),
```
Comment (bnoordhuis) [inline std/node/buffer.ts:19]: If performance matters here: there's probably a steep performance cliff for longer strings.

V8 has a small-numbers-as-strings cache that's not effective for longer strings and I doubt V8 can figure out it doesn't actually need to materialize the indexes as strings.

File: std/node/buffer.ts [hunk 3]
```diff
@@ -431,13 +447,26 @@ export class Buffer extends Uint8Array {
    * encoding. start and end may be passed to decode only a subset of buf.
    */
   toString(encoding = "utf8", start = 0, end = this.length): string {
+    if (start > this.length || end <= 0 || end <= start) {
+      return "";
+    }
+    //Prevent out of bounds
+    start = start < 0 ? 0 : start;
+    end = end > this.length ? this.length : end;
```
Comment (bnoordhuis) [inline std/node/buffer.ts:455]: Don't know if it matters but these let through NaNs.

**Review by bnoordhuis — APPROVED** [review]
LGTM, thanks!

File: std/node/buffer.ts (hunk 1 above)
Comment (benjamingr) [reply std/node/buffer.ts:14]: What about encoding/ascii in deno's code? I'm pretty sure Deno has ASCII encoding/decoding support (I vaguely recall using it)

**Comment by bartlomieju** [comment]
Deno standard library was moved to [denoland/deno_std](https://github.com/denoland/deno_std), please reopen your PR there.

**Pull request closed without merging**